### PR TITLE
Development

### PR DESCRIPTION
**Pull request by MMhaidari**
### We @MMhaidari  @LoviMadaan @Rana-Mudassir worked on the mentioned below Project requirements to complete the Project.

- [ ] You should follow the layout of the wireframes provided. You should personalize the rest of the design including colors, typographies, spacings, etc.
- [ ] Login page and registration page:
- [ ] Should be built with Devise.
- [ ] Food list (only for a group that has less than 3 members):
- [ ] Should display a list of food added by the logged-in user as in the wireframe (display also quantity of a given food).
- [ ] Should lead to a form that allows users to add new food.
Recipes list:
- [ ] Should display a list of recipes created by the logged-in user as in the wireframe.
- [ ] Should lead to recipe details.
 - [ ] If the user is the owner of the recipe, should allow the user to delete it.
Public recipe list:
- [ ] Should display a list of all public recipes ordered by newest as in the wireframe.
- [ ] Should lead to recipe details.
Recipe details:
- [ ] Should display a toggle button that allows for a recipe to be made public or private.
- [ ] If the recipe is public or the user is the owner of the recipe, should display the recipe details as in the wireframe.
- [ ] If the user is the owner of the recipe, should lead to the form that allows the user to add new food.
- [ ] General shopping list view (only for a group that has less than 3 members):
- [ ] Should show the list of food that is missing for all recipes of the logged-in user (compare the list of food for all recipes with the general food list of that user).
- [ ] Should count the total food items and total price of the missing food.
- [ ] Make sure there are no N+1 queries happening.
Create a navigation menu that allows users to open all of the pages you created.
- [ ] Write unit and integration tests

Inventories list:
- [ ] Should display a list of inventories created by the logged-in user as in the wireframe.
- [ ] Should lead to inventory details.
- [ ] If the user is the owner of the inventory, should allow to delete it.
Inventory details:
- [ ] Should display the inventory details as in the wireframe.
- [ ] Should lead to a form that allows users to add new food.
Recipe details:
- [ ] Should have a modal to choose an inventory to create a shopping list with, as in the wireframe.
Inventory shopping list:
- [ ] Should show the list of food that is missing by comparing the food in the recipe with the food in the inventory.
- [ ] Should count the total food items and total price of the missing food.

Thanks for Reviewing 😊

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  # These are extensions that must be enabled in order to support this database
+  enable_extension "plpgsql"
+
+  create_table "foods", force: :cascade do |t|
+    t.string "name"
+    t.string "measurement_unit"
+    t.decimal "price"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.string "quantity_unit"
```
Comment (Shedrack-Sunday) [inline db/schema.rb:23]: - [ ] Kindly note that from the [Entity Relationship Diagram](https://github.com/microverseinc/curriculum-rails/blob/main/recipe-app/images/recipe_app_erd.png?raw=true), you're missing a `User ID` on the `Food` table. Kindly rectify this 👍 

### View:
image

File: app/controllers/recipe_foods_controller.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  def create
+    @recipe = Recipe.find(params[:recipe_id])
+    @new_recipe = @recipe.recipe_foods.new(recipe_food_params)
+    @foods = Food.all
+
+    if @new_recipe.save
+      flash[:notice] = 'Ingredient was added successfully.'
+      redirect_to recipe_path(@recipe.id)
+    else
+      render 'new', status: :unprocessable_entity
```
Comment (Shedrack-Sunday) [inline app/controllers/recipe_foods_controller.rb:16]: - [  ] Nice work in the views of this controller 👏 . However, let's stick to the design to have a close replica of the wireframe. Please compare the images below to adjust your design. You'll agree with me that the remove button should be below the recipe name as provided 👍 

### Template:
image

### View:
image

File: app/views/recipe_foods/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+          <div class="border py-2 px-4">
+            <%= f.select :food_id, @foods.collect { |food| [food.name, food.id] }, :prompt => 'Select an Ingredient', class: 'focus:outline-none', required: true %>
+          </div>
+          <div>
+            <%= f.label :quantity, 'Quantity:' %>
+            <%= f.number_field :quantity, step: :any, required: true, class: 'border py-2 px-4 w-80'%>
+          </div>
+        </div>
+        <div>
+          <%= f.submit "Add Ingredient", class: 'py-3 px-8 bg-sky-500 text-white hover:bg-sky-700 cursor-pointer mt-5'%>
```
Comment (Shedrack-Sunday) [inline app/views/recipe_foods/new.html.erb:19]: - [ ] There seems to be a bug here. When I add an ingredient, it just reloads the page without creating an effect. 
- [ ] Kindly fix this bug to compliment your brillant work so far 👍 

### Error View:
image

File: app/views/inventories/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,35 @@
+<div class="container mx-auto max-w-md">
+  <div class="mb-6 flex justify-end"> 
+    <% if current_user %>
+    <%= link_to "Add inventory", new_inventory_path, class: "bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded mt-4" %>
+    <% end %>
```
Comment (Shedrack-Sunday) [inline app/views/inventories/index.html.erb]: - [ ] You did an awesome job 👏 . However, the plan is to have an exact replica of the format given in the wireframe. I've made a comprehensive review below for you.

### View:
image

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hello @MMhaidari @LoviMadaan and @Rana-Mudassir      👋 

# CHANGES REQUIRED ♻️

Impressive work on the project so far👏
You've done extremely well so far in this project,  However, there are some adjustments you need to make to move to the next project. 💯



## Highlights   
- Good use of GitHub features ✔️   
- Nice work with the rails structure✔️
- Good use of models ✔️
- Impressive and beautiful design and choice of colors 👏  
   
# Required Changes  🌵  

- [ ] You've done well, However, kindly check some key point note I pointed out to meet the requirements before moving to the next project 👍

_Check the comments for a detailed review 

# Optional suggestions 

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing really improves your application.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by MMhaidari** [comment]
# What is added new
- Requested changes added it to the recipe views and inventories
- As I checked with team the in our ends the adding ingredients is working fine 
- As the diagram provided for the project by Micorverse there is no relation between between user and food table below image is for more info 
recipe_erd

File: app/views/inventories/index.html.erb (hunk 4 above)
Comment (Rana-Mudassir) [reply app/views/inventories/index.html.erb]: Resolved now. Thanks 
inventory

File: app/controllers/recipe_foods_controller.rb (hunk 2 above)
Comment (Rana-Mudassir) [reply app/controllers/recipe_foods_controller.rb:16]: Resolved now. Thanks
recioe

File: db/schema.rb (hunk 1 above)
Comment (MMhaidari) [reply db/schema.rb:23]: this is the DML that is provided for as by microverse @Shedrack-Sunday 
recipe_erd

**Review by victorokoroji — APPROVED** [review]
Hi  @MMhaidari, 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
congratulations!  🎉💯🌟 

## Top highlight 🎯

- Requested changes implemented✅ 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/recipe_foods/new.html.erb (hunk 3 above)
Comment (Rana-Mudassir) [reply app/views/recipe_foods/new.html.erb:19]: It's working fine, you can add the ingredient, and will be displayed on the screen as well, you can see the below-attached images.

ingredient

ingredient2

**Pull request merged**